### DISCUSSION TITLE
Any plan to upgrade to vim9?

### DISCUSSION THREAD

**Discussion by ubaldot** [w0rp/ale#4381] [General]
Just wondering:)

**Comment by w0rp** [comment]
Nah, not any time soon. I like to maintain compatibillity with older Vim versions for years and years, and old, old versions of Vim don't support vim9script. Neovim will never have vim9script as a native thing. Good old Vim script is the only way. There aren't really any problems with Vim script in terms of anything like performance I've seen for ALE. At least not anything that can't be overcome pretty easily when I have time to patch whatever problems there are.

vim9script will probably never be generally useful due to the split in the community it causes. Neovim won't ever support it, so it's dead in the water.

**Reply by pbnj** [reply]
Additionally, with Bram's passing, vim9script is likely to get deprioritized as it is expected to be quite a complex undertaking from a development and maintenance standpoint. Bram designed, developed, and maintained the language full-time. The current set of maintainers cannot dedicate this much time or effort to a bespoke language.

See [this thread](https://github.com/vim/vim/discussions/12736#discussioncomment-6651353) for more details.
